### ISSUE TITLE
Unable to install 1.17.1 Forge. Stuck at "Downloading forge installer 100%"

### ISSUE THREAD

**Issue by sylvesterroos** [gorilla-devs/GDLauncher#1008]
**Describe the bug**
Forge will not finish installing. The loading screen/icon is stuck at "Downloading forge installer 100%"

**To Reproduce**
Click on the '+' icon
Try to install a 1.17.1 version of forge
Wait until it gets stuck

**Expected behavior**
The installer should finish successfully like it does with Fabric for example.

**Possible solutions**
N/A

**Screenshots**
![Screenshot_20210727_140451](https://user-images.githubusercontent.com/39904003/127150960-45a6a973-69f3-45a8-9454-4e0a44682a06.png)

**Operating System:**
 - OS: openSUSE
 - Version: Leap 15.3

**Comment by Eskaan** [comment]
This version of forge is A Beta version so please don’t blame us for the installer not working. Most launchers have problems whit that. Please only open issues when you get problems whit releases of forge (or fabric). You can always ask for help on our Discord Server :)

**Comment by Eskaan** [comment]
Also this is a duplicate of #998

**Comment by ramonGonzEdu** [comment]
This is caused because the forgeJSON.install object has the path property set to null instead of an actual path. I don't know where the forgeJSON is got from though

**Comment by ramonGonzEdu** [comment]
If you download the latest forge installer JSON, unzip it (might need to rename), the install_profile.json file has the following line:
```json
{
    "_comment_": [
        "Please do not automate the download and installation of Forge.",
        "Our efforts are supported by ads from the download page.",
        "If you MUST automate this, please consider supporting the project through https://www.patreon.com/LexManos/"
    ],
    "spec": 1,
    "profile": "forge",
    "version": "1.17.1-forge-37.0.12",
    "path": null,
    "minecraft": "1.17.1"
    
}
```

After the minecraft version, there is a large list of deps. the path property is set to null which is not handled on GDLauncher and causes an error on the function `mavenToArray` since the first line calls `s.split` without nullchecking any function parameters. This definitely seems to be an issue with GDLauncher since JSON from a 3rd party is being blindly read without any null checks or sanity checks to prevent this.

Edit: The line in the json that breaks GDL is the `"path": null,`

**Comment by Shad0wyDr3ams** [comment]
this should be fixable via adding this to path line      
 "path": "net.minecraftforge:forge:1.17.1-37.0.12"

the reason i say this is because in every other version of forge before 1.17-1.17.1, that is how the path is named with the exception of the numbers being different per version.

With that being said that would have to be done in every version of forge installer  for gdlauncher to install it .

To my knowledge GDLauncher has no way of allowing custom forge installs... this being said forge itself needs to implement the fix or gd launcher needs to implement a bypass in their launcher so it doesnt read the path line. Or give people the option to do custom forge installs.

**Comment by hypherionmc** [comment]
Fixed in release 1.1.12+
